libclc/amdgcn/lib/cl_khr_int64_extended_atomics/minmax_helpers.ll

```diff
@@ -1,8 +1,4 @@
-#if __clang_major__ >= 7
-target datalayout = "e-p:64:64-p1:64:64-p2:32:32-p3:32:32-p4:64:64-p5:32:32-p6:32:32-i64:64-v16:16-v24:32-v32:32-v48:64-v96:128-v192:256-v256:256-v512:512-v1024:1024-v2048:2048-n32:64-S32-A5"
-#else
-target datalayout = "e-p:32:32-p1:64:64-p2:64:64-p3:32:32-p4:64:64-p5:32:32-i64:64-v16:16-v24:32-v32:32-v48:64-v96:128-v192:256-v256:256-v512:512-v1024:1024-v2048:2048-n32:64"
-#endif
+target datalayout = "e-p:64:64-p1:64:64-p2:32:32-p3:32:32-p4:64:64-p5:32:32-p6:32:32-p7:160:256:256:32-p8:128:128-p9:192:256:256:32-i64:64-v16:16-v24:32-v32:32-v48:64-v96:128-v192:256-v256:256-v512:512-v1024:1024-v2048:2048-n32:64-S32-A5-G1-ni:7:8:9"
```

Comment:
Could we just remove these explicit data layouts, and rely on `llvm-link` giving us the correct one from a source-compiled object, @arsenm?

Reply:
It would work in this situation, but technically this is a toothpaste-out-of-the-tube situation.

But really we should stop using iR for these. I don't know why these specifically are in the first place, but I'm guessing we didn't have complete builtin coverage in clang for atomics in the past. These should all be implementable with the atomic builtins these days

Reply:
Hah, yeah, true. Though I think there'll always be a need for IR - at least, I'm not sure we should ban it outright. I don't think I'll find the time to convert these AMDGPU files to OpenCL C any time soon.

So if scrubbing data layouts works in principle I'd lean towards going for it, at least in the short/medium term. It's not a great look that these warnings are coming out of libclc.

That said, just grouping the IR files together at the start of the link line eliminates the bulk of warnings so if you want to keep the data layouts in, we're also in a better position than before this PR.

Reply:
This also isn't amdgpu specific. I don't know why this is in the target directory

Reply:
Good point, yes. At least now there's no data layout we can more easily make it generic. It looks like we've got a couple almost identical copies of this file downstream, so I'll refactor that in due course.